ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryExecutor.java

```diff
@@ -510,17 +529,22 @@ private ProcessingLogger getUncaughtExceptionProcessingLogger(final QueryId quer
   private static TransientQueryQueue buildTransientQueryQueue(
       final Object buildResult,
       final OptionalInt limit,
-      final boolean excludeTombstones
+      final boolean excludeTombstones,
+      final Optional<ImmutableMap<TopicPartition, Long>> endOffsets
   ) {
+    // If we are supposed to stop at the endOffsets, but we didn't get any, then we should
```

Comment:
In what scenario are we not getting any endOffsests even though we are supposed to?

Reply:
It's a bit weird, but if a partition is empty, then we just won't get an "end offset" for it at all. So, if _all_ the partitions are empty, then the whole map will be empty. In this case, it just means we're supposed to stop at the "end" of an empty topic. Aka, we should return zero results.